test/index.js

```diff
@@ -1170,6 +1170,29 @@ describe('deepEqual()', function () {
         expect(Hoek.deepEqual(a, b, { prototype: false})).to.be.true();
         done();
     });
+
+    it('compares complex objects', function (done) {
+
+        var a = {
+            provider: 'custom',
+            token: '456',
+            expiresIn: 3600,
+            refreshToken: undefined,
+            query: {},
+            profile: {
+                id: '1234567890',
+                name: {
+                    first: 'steve',
+                    last: 'smith'
+                },
+                headline: 'Master of the universe',
+                raw: profile
```

Comment:
This isn't broken due to Deep Equal in the test currently. It is broken because this self-references itself before it is created. Change this to a primitive or create the profile object and then set raw to it.